### DISCUSSION TITLE
How important is it for you that the project doesn't use JS?

### DISCUSSION THREAD

**Discussion by nigini** [tinyBlogPub/microblog.pub#5] [Q&A]
I personally don't have problems with JS, especially when it is used to improve User Experience.
For example, I'd love to have a post editor like this: https://github.com/sparksuite/simplemde-markdown-editor

**Comment by 3b0b** [comment]
It is very important to me that the project does not require JS for my readers. I would not necessarily mind some progressive enhancement, nor some JS-enabled convenience features in the admin/composition section, especially as long as micropub still works.

**Reply by nigini** [reply]
Thanks for your input @3b0b. That reinforces some thoughts I had about the multiple requirements that MicroBlog.pub has been juggling with: micropub server, micropub client, activitypub server, activitypub client, blogging platform, blogging site...

I have been wondering about a v3 that has all of the above as modules, and you can pick whatever serves you best. I have started [another discussion on this](https://github.com/orgs/microblog-pub/discussions/6)!

**Comment by JD557** [comment]
On top of the project not requiring JS for readers, it's also important to me that the Admin interface still works on [KaiOS](https://www.kaiostech.com/) (Yes, I know that's a pain to test...), which does support most JavaScript.

Having said that, looking at the demo in https://simplemde.com/ with JavaScript disabled, it seems to fallback to a normal `<textarea>`, so this seems like an easy win (the current editor already uses JS for the emojis anyway).

**Reply by nigini** [reply]
This is a great reminder, that people like no-JS because of low-resource devices!!!
Thanks for testing the SimpleMDE. Good to know it would not compromise functionalities for no-JS devices!

**Reply by nigini** [reply]
Also, @JD557, out of curiosity, did you mean to say that the KaiOS DOES or DOES NOT support JS?

**Reply by JD557** [reply]
It does support it, but I think a lot of devices don't receive updates (stuck in KaiOS 2, IIRC), so some modern features might not be supported.

**Comment by 1dom** [comment]
On this topic, is there a particular reason for htmx being added: https://github.com/microblog-pub/microblog.pub/commit/3de398bf137f397d1fadfa844a7519bdd964791a 

It stands out as being the only js library outside of the hand coded common js and is about 50x bigger than them both. It's also static minified v1 of htmx which is now on v2.

**Reply by JD557** [reply]
> is there a particular reason for htmx being added

I added it on my fork for personal reasons, but maybe it doesn't make sense to add it here. It's probably better to remove it.

**Reply by nigini** [reply]
Created issue #12 on that!

**Comment by nigini** [comment]
Looking through this whole discussion, it seems like adding a nicer MarkDown editor to the admin page is acceptable as long as it: 
1. does not break the page on browsers that are blocking or do not support JS;
2. does not mess up with micropub functionalities.

Does it make sense to have separate admin pages so that a user can opt to use a simplified experience instead of a more modern UX?

**Reply by 3b0b** [reply]
I would expect a good "nicer MarkDown editor" to degrade gracefully in the absence of javascript, personally. If it doesn't, it's probably not one I'd be in favor of using. So I wouldn't really want a separate page, at least not based on what I imagine the experience being like.
